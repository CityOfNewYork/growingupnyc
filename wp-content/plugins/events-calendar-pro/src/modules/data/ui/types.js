/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY`;

export const TOGGLE_RULE_PANEL_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_RULE_PANEL_VISIBILITY`;
export const HIDE_RULE_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/HIDE_RULE_PANEL`;

export const TOGGLE_RULE_PANEL_EXPAND = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_RULE_PANEL_EXPAND`;
export const EXPAND_RULE_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/EXPAND_RULE_PANEL`;

export const TOGGLE_EXCEPTION_PANEL_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_EXCEPTION_PANEL_VISIBILITY`;
export const HIDE_EXCEPTION_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/HIDE_EXCEPTION_PANEL`;

export const TOGGLE_EXCEPTION_PANEL_EXPAND = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_EXCEPTION_PANEL_EXPAND`;
export const EXPAND_EXCEPTION_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/EXPAND_EXCEPTION_PANEL`;
