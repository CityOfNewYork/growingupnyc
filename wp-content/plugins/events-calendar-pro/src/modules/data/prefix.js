export const PREFIX_EVENTS_PRO_STORE = '@@MT/EVENTS-PRO';
