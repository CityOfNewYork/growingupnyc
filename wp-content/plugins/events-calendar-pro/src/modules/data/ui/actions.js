/**
 * Internal dependencies
 */
import * as types from './types';

export const toggleRepeatBlocksVisibility = () => ( {
	type: types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY,
} );

export const toggleRulePanelVisibility = () => ( {
	type: types.TOGGLE_RULE_PANEL_VISIBILITY,
} );

export const hideRulePanel = () => ( {
	type: types.HIDE_RULE_PANEL,
} );

export const toggleExceptionPanelVisibility = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_VISIBILITY,
} );

export const toggleRulePanelExpand = () => ( {
	type: types.TOGGLE_RULE_PANEL_EXPAND,
} );

export const expandRulePanel = () => ( {
	type: types.EXPAND_RULE_PANEL,
} );

export const hideExceptionPanel = () => ( {
	type: types.HIDE_EXCEPTION_PANEL,
} );

export const expandExceptionPanel = () => ( {
	type: types.EXPAND_EXCEPTION_PANEL,
} );

export const toggleExceptionPanelExpand = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_EXPAND,
} );

