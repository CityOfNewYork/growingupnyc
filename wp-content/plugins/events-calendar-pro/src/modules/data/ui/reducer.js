/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	isRepeatBlockVisible: false,
	isRulePanelVisible: false,
	isExceptionPanelVisible: false,
	isRulePanelExpanded: false,
	isExceptionPanelExpanded: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY:
			return {
				...state,
				isRepeatBlockVisible: ! state.isRepeatBlockVisible,
			};
		case types.TOGGLE_RULE_PANEL_VISIBILITY:
			return {
				...state,
				isRulePanelVisible: ! state.isRulePanelVisible,
			};
		case types.HIDE_RULE_PANEL:
			return {
				...state,
				isRulePanelVisible: false,
			};
		case types.TOGGLE_EXCEPTION_PANEL_VISIBILITY:
			return {
				...state,
				isExceptionPanelVisible: ! state.isExceptionPanelVisible,
			};
		case types.HIDE_EXCEPTION_PANEL:
			return {
				...state,
				isExceptionPanelVisible: false,
			};
		case types.TOGGLE_RULE_PANEL_EXPAND:
			return {
				...state,
				isRulePanelExpanded: ! state.isRulePanelExpanded,
			};
		case types.EXPAND_RULE_PANEL:
			return {
				...state,
				isRulePanelExpanded: true,
			};
		case types.TOGGLE_EXCEPTION_PANEL_EXPAND:
			return {
				...state,
				isExceptionPanelExpanded: ! state.isExceptionPanelExpanded,
			};
		case types.EXPAND_EXCEPTION_PANEL:
			return {
				...state,
				isExceptionPanelExpanded: true,
			};
		default:
			return state;
	}
};
