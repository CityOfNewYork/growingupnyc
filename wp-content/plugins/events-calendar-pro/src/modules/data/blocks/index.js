/**
 * Internal dependencies
 */
import * as constants from './constants';
import * as recurring from './recurring';
import reducer from './reducer';

export default reducer;
export { constants };
export { recurring };
