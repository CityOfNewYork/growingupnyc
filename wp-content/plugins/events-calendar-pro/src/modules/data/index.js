/**
 * External dependencies
 */
import reducer from './reducers';

import { actions, constants } from '@moderntribe/common/data/plugins';
import { store } from '@moderntribe/common/store';
import initSagas from './sagas';
import * as blocks from './blocks';
import '@moderntribe/events-pro/data/sagas';

const { EVENTS_PRO_PLUGIN } = constants;

export const initStore = () => {
	const { dispatch, injectReducers } = store;
	initSagas();

	dispatch( actions.addPlugin( EVENTS_PRO_PLUGIN ) );
	injectReducers( { [ EVENTS_PRO_PLUGIN ]: reducer } );
};

export const getStore = () => store;

export { blocks };
