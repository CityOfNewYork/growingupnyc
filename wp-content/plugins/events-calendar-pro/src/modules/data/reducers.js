/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import ui from './ui';
import blocks from './blocks';
import status from './status';

export default combineReducers( {
	ui,
	blocks,
	status,
} );
