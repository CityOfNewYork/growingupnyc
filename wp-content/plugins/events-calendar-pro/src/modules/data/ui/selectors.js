/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
export const isRulePanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelVisible
);
export const isExceptionPanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelVisible
);
export const isRulePanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelExpanded
);
export const isExceptionPanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelExpanded
);
