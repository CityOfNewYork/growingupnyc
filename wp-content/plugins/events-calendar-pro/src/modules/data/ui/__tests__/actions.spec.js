/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/ui';

describe( '[STORE] - UI actions', () => {
	test( 'toggleRepeatBlocksVisibility', () => {
		expect( actions.toggleRepeatBlocksVisibility() ).toMatchSnapshot();
	} );

	test( 'toggleRulePanelVisibility', () => {
		expect( actions.toggleRulePanelVisibility() ).toMatchSnapshot();
	} );

	test( 'hideRulePanel', () => {
		expect( actions.hideRulePanel() ).toMatchSnapshot();
	} );

	test( 'toggleExceptionPanelVisibility', () => {
		expect( actions.toggleExceptionPanelVisibility() ).toMatchSnapshot();
	} );

	test( 'toggleRulePanelExpand', () => {
		expect( actions.toggleRulePanelExpand() ).toMatchSnapshot();
	} );

	test( 'expandRulePanel', () => {
		expect( actions.expandRulePanel() ).toMatchSnapshot();
	} );

	test( 'hideExceptionPanel', () => {
		expect( actions.hideExceptionPanel() ).toMatchSnapshot();
	} );

	test( 'expandExceptionPanel', () => {
		expect( actions.expandExceptionPanel() ).toMatchSnapshot();
	} );

	test( 'toggleExceptionPanelExpand', () => {
		expect( actions.toggleExceptionPanelExpand() ).toMatchSnapshot();
	} );
} );
