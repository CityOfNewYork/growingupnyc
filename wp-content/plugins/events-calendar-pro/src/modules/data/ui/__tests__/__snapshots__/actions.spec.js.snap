// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI actions expandExceptionPanel 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/EXPAND_EXCEPTION_PANEL",
}
`;

exports[`[STORE] - UI actions expandRulePanel 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/EXPAND_RULE_PANEL",
}
`;

exports[`[STORE] - UI actions hideExceptionPanel 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/HIDE_EXCEPTION_PANEL",
}
`;

exports[`[STORE] - UI actions hideRulePanel 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/HIDE_RULE_PANEL",
}
`;

exports[`[STORE] - UI actions toggleExceptionPanelExpand 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/TOGGLE_EXCEPTION_PANEL_EXPAND",
}
`;

exports[`[STORE] - UI actions toggleExceptionPanelVisibility 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/TOGGLE_EXCEPTION_PANEL_VISIBILITY",
}
`;

exports[`[STORE] - UI actions toggleRepeatBlocksVisibility 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY",
}
`;

exports[`[STORE] - UI actions toggleRulePanelExpand 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/TOGGLE_RULE_PANEL_EXPAND",
}
`;

exports[`[STORE] - UI actions toggleRulePanelVisibility 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/TOGGLE_RULE_PANEL_VISIBILITY",
}
`;
